File: ports/java/src/core/io/BcsTextStream.java
package bcs.core.io;

import bcs.core.global.BcsString;
import java.io.*;

// BcsTextStream provides a convenient interface for reading and writing text.
public class BcsTextStream {
    private BufferedReader reader;
    private PrintWriter writer;

    public BcsTextStream(InputStream in, OutputStream out) {
        if (in != null) {
            this.reader = new BufferedReader(new InputStreamReader(in));
        }
        if (out != null) {
            this.writer = new PrintWriter(out, true);
        }
    }

    public BcsString readLine() throws IOException {
        if (this.reader != null) {
            return new BcsString(this.reader.readLine());
        }
        return new BcsString("");
    }

    public void writeString(BcsString str) {
        if (this.writer != null) {
            this.writer.print(str.toString());
            this.writer.flush();
        }
    }
}

File: ports/csharp/src/core/io/BcsTextStream.cs
using System.IO;
using BCS.Core.Global;

namespace BCS.Core.IO {
    // BcsTextStream provides a convenient interface for reading and writing text.
    public class BcsTextStream {
        private StreamReader _reader;
        private StreamWriter _writer;

        public BcsTextStream(Stream stream) {
            _reader = new StreamReader(stream);
            _writer = new StreamWriter(stream) { AutoFlush = true };
        }

        public BcsString ReadLine() {
            return new BcsString(_reader.ReadLine());
        }

        public BcsString ReadAll() {
            return new BcsString(_reader.ReadToEnd());
        }

        public void WriteString(BcsString str) {
            _writer.Write(str.ToString());
        }
    }
}
